core/chain_makers.go

```diff
@@ -340,7 +366,74 @@ func (cr *fakeChainReader) GetGasUsedInChain(block *types.Block, length int) int
 func (cr *fakeChainReader) GetExternalBlocks(header *types.Header) ([]*types.ExternalBlock, error) {
 	return nil, nil
 }
-
 func (cr *fakeChainReader) GetLinkExternalBlocks(header *types.Header) ([]*types.ExternalBlock, error) {
 	return nil, nil
 }
+
+// GenerateNetwork creates a network of n blocks in c contexts. The first block's
+// parent will be the provided parent. db is used to store intermediate states
+// and should contain the parent's state trie.
+
+// configs should contain the respective config settings being tested. This must
+// include a valid slice of Prime, Region, and Zone mining locations. This is
+// necessary for testing interchain linkages through coincident blocks as well
+// as transactions flowing through them. The array of ints in c represent
+// the intended order of contexts to be mined.
+
+// The generator function is called with a new block generator for every
+// block. Any transactions and uncles added to the generator become part of the
+// block. If gen is nil, the blocks will be empty and their coinbase will
+// be the zero address.
+
+// Blocks created by GenerateNetwork do not contain valid proof of work values.
+// Inserting them into BlockChain requires use of a TestPoW in order to test and
+// verify the correct operation of the Proof-of-Work algorithm.
+func GenerateNetwork(genesisCheck bool, config *params.ChainConfig, parent *types.Block, order int, startNumber [3]int, n int, engine consensus.Engine, db ethdb.Database, gen func(int, *BlockGen)) ([]*types.Block, []types.Receipts) {
+	// initialize blocks and receipts
+	blocks, receipts := make(types.Blocks, n), make([]types.Receipts, n)
+
+	// associate chainreaders and configs for correct block context placement
+	chainreader := fakeChainReader{config: config}
+
+	genblock := func(genCheck bool, i int, parent *types.Block, order int, statedb *state.StateDB, chainreader fakeChainReader, config params.ChainConfig) (*types.Block, types.Receipts) {
```

Comment:
Somewhere in here, you need to actually mine the block to reach a specified difficulty order. The general overview should be something like:
1) construct header from a zone's perspective
2) mine that header
3) if the mined header > order: repeat from step 2
4) commit the mined block to the db